### PR TITLE
chore: Add the bitcoin canister's candid interface to its metadata section

### PR DESCRIPTION
**Pull request by dsarlis**

### CODE REVIEW

**Review by dragoljub-duric — APPROVED** [review]

File: scripts/build-canister.sh [hunk 1]
```diff
@@ -34,6 +34,14 @@ if [[ "$STATUS" -eq "0" ]]; then
     ./target/bin/ic-wasm \
     "./target/$TARGET/release/$CANISTER.wasm" \
     -o "./target/$TARGET/release/$CANISTER.wasm" shrink
+
+    if [[ "$CANISTER" == "ic-btc-canister" ]]; then
+    	./target/bin/ic-wasm \
+    	"./target/$TARGET/release/$CANISTER.wasm" \
+    	-o "./target/$TARGET/release/$CANISTER.wasm" \
+    	metadata candid:service -f "$SCRIPT_DIR/../canister/candid.did" -v public
```
Comment (ielashi) [inline scripts/build-canister.sh:42]: Is there anything we can run locally to see the metadata? I wonder how one can verify that this command works.

**Review by ielashi — APPROVED** [review]

File: scripts/build-canister.sh (hunk 1 above)
Comment (dsarlis) [reply scripts/build-canister.sh:42]: Oh yeah, I forgot to add this. If you build the canisters using docker; extract the bitcoin canister's Wasm, then unzip it and run `ic-wasm <bitcoin_canister_wasm> info` you can see that there's a public metadata section called `candid:service`.

Comment (dsarlis) [reply scripts/build-canister.sh:42]: I added a workflow to check this which is even better I think.

**Comment by dsarlis** [comment]
@ielashi @dragoljub-duric PTAL for the github workflow that I added. Should be quite straightforward but don't want to push without a second pair of eyes at least.

File: e2e-tests/bitcoin-canister-metadata.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# Run dfx stop if we run into errors.
+trap "dfx stop" EXIT SIGINT
+
+dfx start --background --clean
+
+# Deploy the bitcoin canister, setting the blocks_source to be the source above.
+dfx deploy --no-wallet bitcoin --argument "(record {
+  stability_threshold = opt 2;
+  network = opt variant { regtest };
+})"
```
Comment (ielashi) [inline e2e-tests/bitcoin-canister-metadata.sh:16]: You don't actually need any arguments here.

```suggestion
# Deploy the bitcoin canister
dfx deploy --no-wallet bitcoin --argument "(record { })"
```

**Review by ielashi — APPROVED** [review]
LGTM!

**Pull request merged**